src/ClientHandle.cpp

```diff
@@ -2187,6 +2200,13 @@ void cClientHandle::SendDisconnect(const AString & a_Reason)
 		LOGD("Sending a DC: \"%s\"", StripColorCodes(a_Reason).c_str());
 		m_Protocol->SendDisconnect(a_Reason);
 		m_HasSentDC = true;
+
+		NetworkTick();
+		cTCPLinkPtr Link(m_Link);  // Grab a copy of the link in a multithread-safe way
```

Comment:
Could someone explain why here a copy of m_Link must be made, whereas in NetworkTick above, m_Link can be acted upon directly?